### PR TITLE
Small optimizations for os.go

### PR DESCRIPTION
**Pull request by Limero**

### CODE REVIEW

File: os.go [hunk 1]
```diff
@@ -189,42 +190,39 @@ func isHidden(f os.FileInfo, path string, hiddenfiles []string) bool {
...
 		}
+		hidden = !strings.HasPrefix(pattern, "!")
 	}
 	return hidden
 }
 
 func userName(f os.FileInfo) string {
 	if stat, ok := f.Sys().(*syscall.Stat_t); ok {
-		uid := fmt.Sprint(stat.Uid)
+		uid := strconv.FormatUint(uint64(stat.Uid), 10)
```
Comment (Limero) [inline os.go:203]: This should be faster, see https://stackoverflow.com/questions/24886015/how-to-convert-uint32-to-string#comment94954590_24886089

File: os.go [hunk 2]
```diff
@@ -189,42 +190,39 @@ func isHidden(f os.FileInfo, path string, hiddenfiles []string) bool {
...
 	for _, pattern := range hiddenfiles {
 		matched := matchPattern(strings.TrimPrefix(pattern, "!"), f.Name(), path)
-		if strings.HasPrefix(pattern, "!") && matched {
-			hidden = false
-		} else if matched {
-			hidden = true
+		if !matched {
+			continue
 		}
+		hidden = !strings.HasPrefix(pattern, "!")
```
Comment (joelim-work) [inline os.go:196]: I think you can simplify this even further:

```go
if matchPattern(strings.TrimPrefix(pattern, "!"), f.Name(), path) {
	hidden = !strings.HasPrefix(pattern, "!")
}
```

**Review by joelim-work — COMMENTED** [review]
Thanks for the changes, I just had a small comment but looks fine otherwise.

File: os.go (hunk 2 above)
Comment (Limero) [reply os.go]: That's even nicer, thanks!

**Review by joelim-work — APPROVED** [review]

**Pull request merged**